device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/DeviceIO.java

```diff
@@ -145,26 +139,37 @@
      */
     void open() throws IOException
     {
-        /* Codes_SRS_DEVICE_IO_21_007: [If the client is already open, the open shall do nothing.] */
-        if (this.state == IotHubClientState.OPEN)
-        {
-            return;
-        }
+        open(true);
+    }
 
-        /* Codes_SRS_DEVICE_IO_21_012: [The open shall open the transport to communicate with an IoT Hub.] */
-        /* Codes_SRS_DEVICE_IO_21_015: [If an error occurs in opening the transport, the open shall throw an IOException.] */
-        try
-        {
-            this.transport.open(deviceClientConfigs);
-        }
-        catch (DeviceClientException e)
+    void open(boolean openTransport) throws IOException
+    {
+        synchronized (stateLock)
         {
-            throw new IOException("Could not open the connection", e);
-        }
+            /* Codes_SRS_DEVICE_IO_21_007: [If the client is already open, the open shall do nothing.] */
+            if (this.state == IotHubClientState.OPEN)
+            {
+                return;
+            }
 
-        /* Codes_SRS_DEVICE_IO_21_014: [The open shall schedule receive tasks to run every receivePeriodInMilliseconds milliseconds.] */
-        /* Codes_SRS_DEVICE_IO_21_016: [The open shall set the `state` as `CONNECTED`.] */
-        commonOpenSetup();
+            if (openTransport)
+            {
+                /* Codes_SRS_DEVICE_IO_21_012: [The open shall open the transport to communicate with an IoT Hub.] */
+                /* Codes_SRS_DEVICE_IO_21_015: [If an error occurs in opening the transport, the open shall throw an IOException.] */
+                try
+                {
+                    this.transport.open(deviceClientConfigs);
+                }
+                catch (DeviceClientException e)
+                {
+                    throw new IOException("Could not open the connection", e);
+                }
+            }
+
+            /* Codes_SRS_DEVICE_IO_21_014: [The open shall schedule receive tasks to run every receivePeriodInMilliseconds milliseconds.] */
+            /* Codes_SRS_DEVICE_IO_21_016: [The open shall set the `state` as `CONNECTED`.] */
+            commonOpenSetup();
+        }
```

Comment:
is it ok to call commonOpenSetup with openTransport == false>?

Reply:
commonOpenSetup doesn't do anything to the transport layer, so it wouldn't be useful as a parameter regardless